### DISCUSSION TITLE
IDs in a front end application

### DISCUSSION THREAD

**Discussion by soundsmagic** [leolabs/ableton-js#55] [Q&A] [answered]
Hi, got another question!
In the React front end I'm working on, I'll need some form of IDs to keep track of things when stuff is updated (either there or in Ableton, since I'm using Websockets). I discovered the raw.id property, but when I tried it out, it seemed that the ID changed whenever the object at hand changed. (I guess it has something to do with the save_obj method in Interface, and since the object has changed it gets a new ID from the id() function - if I'm not thinking totally backwards)

Is it still the best option to use this ID and just update it in the front end when it changes, or is there another smoother way?

**Comment by leolabs** [comment, marked as answer]
Unfortunately, Live doesn't offer a stable ID for its objects, but maybe you can use another value to uniquely identify them and keep a map of custom_id -> object which allows you to reference it.

This is how I do it in [AbleSet](https://ableset.app) with locators, for example. Each locator has a unique position that I can use as its key to reference it in the web app and to store some additional metadata on the server-side. Of course, when the position of a marker changes, the code has to make sure that the reference stays the same even though its key changed.

If you don't want to invest the time to write this, just using the `raw.id` property should work as well. Even if the ID of the object changes, you can still reference it using its old ID since it stays in the control surface plugin's map of ID -> Object forever.

**Reply by soundsmagic** [reply]
Thanks once again for your input Leo! I'll think about how to best find a key-value for my frontend objects, or solve it with raw.id in some way...
